core-rust/mesh-api-server/src/mesh_api/handlers/construction_metadata.rs

```diff
@@ -9,7 +9,7 @@ pub(crate) async fn handle_construction_metadata(
 ) -> Result<Json<models::ConstructionMetadataResponse>, ResponseError> {
     assert_matching_network(&request.network_identifier, &state.network)?;
 
-    let database = state.state_manager.database.snapshot();
+    let database = state.state_manager.database.access_direct();
```

Comment:
Can we explain why this is safe in a comment?

Reply:
Decided to revert to `snapshot()` as per [this conversation](https://rdxworks.slack.com/archives/C071C6DB8TG/p1733841098517859?thread_ts=1732877188.486819&cid=C071C6DB8TG)